### ISSUE TITLE
[BUG] VMs on a harvester node can't communicate with the nodes mgmt IP or kube-vip virtual IP

### ISSUE THREAD

**Issue by 2stacks** [harvester/harvester#1312]
**Describe the bug**
A VM running on a Harvester node is unable to communicate with that nodes management interface.  This prevents running an "external" Rancher cluster on VMs inside Harvester and using that Rancher installation's node driver to provision new nodes on the Harvester cluster.

**To Reproduce**
Simple Test
1. Deploy a VM on Harvester
2. From the VM, ping the Harvester node's management IP and kube-vip IP (this works)
3. From the VM, curl the Harvester node's API using the management IP (this fails)
```
curl -k https://<harvester_ip>:8443/k8s/clusters/local/apis/harvesterhci.io/v1beta1/settings/server-version 
```
4. From the VM, curl the Harvester node's API using the kube-vip IP (this fails)
```
curl -k https://<harvester_ip>:8443/k8s/clusters/local/apis/harvesterhci.io/v1beta1/settings/server-version
``` 

Full Steps
1. Deploy a single node harvester cluster
2. Deploy a single node Rancher server
3. Configure Rancher's Harvester node driver using Harvester cluster created in step 1.
4. Attempt to deploy cluster.  Cluster creation fails with error `Error with pre-create check: "Get \"https://<harvester_ip>:8443/k8s/clusters/local/apis/harvesterhci.io/v1beta1/settings/server-version\": dial tcp <harvester_ip>:8443: i/o timeout"; Timeout waiting for ssh key`

**Expected behavior**
Since VMs are using a separate interface from that used for node management, the host should be able to pass this traffic.

**Support bundle**
[support_bundle.zip](https://github.com/harvester/harvester/files/7246880/support_bundle.zip)

**Environment:**
 - Harvester ISO version: `v0.3.0-preview1`
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):  `Baremetal Dell R610`

**Additional context**
ICMP traffic forwards as expected.  Only TCP traffic (443, 8443) was tested.  No sure if this also impacts UDP.

**Comment by DillonN** [comment]
Still running into this issue in 1.1.1. Also I notice it in the reverse, i.e. a harvester node can't reach 443 of guest rancher VMs running on it. Both directions work fine when the VM is on a different node.

**Comment by w13915984028** [comment]
I did a test on Harvester v1.1.2, single-node cluster:

```
baremetal HOST-OS, SUSE SLE15 ------------KVM VM for Harvester NODE -------------------------guest VM, ubuntu, as workload
                                           192.168.122.112                                                    10.52.0.129
```

```
From `10.52.0.129`, it ping `192.168.122.112`, works.
If curl certain ports:
https 8443: Connection refused
htttps 443 (Harvester https webui) : not authorized
https 6443 (k8s): Unauthorized

http 8000 (a simple python http server, run on Harvester): works
```

The failure of https 443, 6443, are due to lack of tls related certification files.

```
ubuntu@vm66:~$ curl -k https://192.168.122.112:8443/k8s/clusters/local/apis/harvsterhci.io/v1beta1/settings/server-version
curl: (7) Failed to connect to 192.168.122.112 port 8443: Connection refused

ubuntu@vm66:~$ curl -k https://192.168.122.112:443/k8s/clusters/local/apis/harvesterhci.io/v1beta1/settings/server-version
not authorized

ubuntu@vm66:~$ curl -k https://192.168.122.112:6443/k8s/clusters/local/apis/harvesterhci.io/v1beta1/settings/server-version
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "Unauthorized",
  "reason": "Unauthorized",
  "code": 401
ubuntu@vm66:~$

ubuntu@vm66:~$ curl  http://192.168.122.112:8000
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<title>Directory listing for /</title>
</head>
<body>
<h1>Directory listing for /</h1>
```

<img width="571" alt="image" src="https://github.com/harvester/harvester/assets/31133476/06a10e5e-2a42-4462-a89b-adf37fa56b70">


@2stacks @DillonN:

what's your CURL error code when:
```
From the VM, curl the Harvester node's API using the management IP (this fails)
,,,
From the VM, curl the Harvester node's API using the kube-vip IP (this fails)
```

**Comment by DillonN** [comment]
@w13915984028 thanks for looking into it. The error I see is just a timeout, like there's nothing listening on the other end. Here's the full output for the different permutations I was trying:

1. Rancher node A -> harvester node it's running on
```
$ curl -k https://10.64.0.20
curl: (28) Failed to connect to 10.64.0.20 port 443 after 129468 ms: Connection timed out
```
2. Rancher node A -> other harvester node
```
$ curl -k https://10.64.0.182
<a href="/dashboard/">Found</a>.
```
3. Rancher node B -> harvester node it's running on
```
$ curl -k https://10.64.0.182
curl: (28) Failed to connect to 10.64.0.182 port 443 after 129404 ms: Connection timed out
```
4. Rancher node B -> other harvester node
```
$ curl -k https://10.64.0.20
<a href="/dashboard/">Found</a>.
```

I see the same results when I do the test in reverse (harvester -> rancher). Also note I see the same when testing http/port 80. Before I noticed the problem was only when talking to the host node, I spent a good while looking into firewall rules and didn't see anything at issue there. 

However, I was able to find a workaround to get Rancher <-> Harvester management working. Turns out the two Rancher cluster agent pods were running on the one Harvester node that had the VIP. After redistributing them across two nodes, the communication is working properly.

**Comment by ibrokethecloud** [comment]
This specific issue was related to v0.3.0

Networking setup has undergone significant changes since v1.1.0. 

Closing this issue, if you notice similar issues on v1.1.x onwards.

**Comment by DillonN** [comment]
@ibrokethecloud I am using Harvester 1.1.2. Are you saying the issue will be fixed in a future 1.1.x version?

**Comment by w13915984028** [comment]
@DillonN : Was your VM using VLAN network, are you still able to reproduce this ?

@mingshuoqiu  Per @DillonN descritption, his cluster is on v1.1.2, could the root cause be similar to https://github.com/harvester/harvester/issues/4359 ? thanks.

**Comment by mingshuoqiu** [comment]
The communication between different node should be on `flannel` which is basically VxLAN. It would be better to run `tcpdump` on both side to check the packet content. I'll need to know what the network topology is and then trying to identify the problem.

**Comment by DillonN** [comment]
@w13915984028 @mingshuoqiu I just retested with the methodology I posted earlier and do not have this issue anymore. Aside from being on Harvester 1.2.1 now, the other thing I think might have resolved this is migrating to use the cluster networks system that was added in 1.1.x. At the time of my last comment I had only recently updated to 1.1.2 and not yet fully migrated to the new network config system.
